lib/std/special/compiler_rt/os_version_check.zig

```diff
@@ -0,0 +1,76 @@
+// Ported from llvm-project 13.0.0 d7b669b3a30345cfcdb2fde2af6f48aa4b94845d
+//
+// https://github.com/llvm/llvm-project/blob/llvmorg-13.0.0/compiler-rt/lib/builtins/os_version_check.c
+
+// The compiler generates calls to __isPlatformVersionAtLeast() when Objective-C's @available
+// function is invoked.
+//
+// Old versions of clang would instead emit calls to __isOSVersionAtLeast(), which is still
+// supported in clang's compiler-rt implementation today in case anyone tries to link an object file
+// produced with an old clang version. This requires dynamically loading frameworks, parsing a
+// system plist file, and generally adds a fair amount of complexity to the implementation and so
+// our implementation differs by simply removing that backwards compatability support. We only use
+// the newer codepath, which merely calls out to the Darwin _availability_version_check API which is
+// available on macOS 10.15+, iOS 13+, tvOS 13+ and watchOS 6+.
+
+inline fn constructVersion(major: u32, minor: u32, subminor: u32) u32 {
+    return ((major & 0xffff) << 16) | ((minor & 0xff) << 8) | (subminor & 0xff);
+}
+
+// Darwin-only
+pub fn __isPlatformVersionAtLeast(platform: u32, major: u32, minor: u32, subminor: u32) callconv(.C) i32 {
+    return if (_availability_version_check(1, &[_]dyld_build_version_t{
+        .{ .platform = platform, .version = constructVersion(major, minor, subminor) },
+    })) 1 else 0;
+}
+
+// TODO(future): If anyone wishes to build Objective-C code for Android, we would need to port this
```

Comment:
IMO this can be removed as it may get out of sync with the upstream impl by the time somebody tries to implement it.